compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -285,18 +173,77 @@ impl SirFuncCx<'tcx> {
             mir.basic_blocks().len()
         ];
 
+        // There will be at least as many locals in the SIR as there are in the MIR.
```

Comment:
There can be more?

Reply:
Yes. We no longer have a one-to-one mapping between SIR and MIR locals due to the way we decompose projections. We emit intermediate stores in certain conditions, which means we may have more locals than the corresponding MIR.

Reply:
OK.